src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Tests/IntegrationTests/AdminServiceTests.cs

```diff
@@ -0,0 +1,140 @@
+        {
+        }
+
+        [Fact]
+        public void Gets_all_tests()
+        {
+            var testList = GetTestList();
+            using (var context = new ApplicationDbContext(this.ContextOptions))
+            {
+                context.AddRange(testList);
+                context.SaveChanges();
+            }
+
+            using (var context = new ApplicationDbContext(this.ContextOptions))
+            {
+                var adminService = PrepareSUT(context);
+
+                var result = adminService.GetAllTests();
+
+                result.Should().BeOfType<List<TestDto>>();
+                result.Count.Should().Be(testList.Count);
+            }
+        }
+
+        [Fact]
+        public void Gets_all_test_answers()
+        {
+            var testDto = GetTestDto();
+            using (var context = new ApplicationDbContext(this.ContextOptions))
+            {
+                var adminService = PrepareSUT(context);
+
+                adminService.AddTest(testDto);
+            }
+
+            using (var context = new ApplicationDbContext(this.ContextOptions))
+            {
+                var result = context.Tests.SingleOrDefault(t => t.Id == testDto.Id);
+                result.HashedAnswers.Count.Should().Be(3);
+            }
+        }
+
+        [Fact]
+        public void Updates_test_dates()
+        {
+            var test = GetTest();
+            var minutes = 30;
+            var minutesString = "30";
+            using (var context = new ApplicationDbContext(this.ContextOptions))
+            {
+                context.Tests.Add(test);
+                context.SaveChanges();
+            }
+
+            using (var context = new ApplicationDbContext(this.ContextOptions))
+            {
+                var adminService = PrepareSUT(context);
+                adminService.UpdateTestDates(test.Id, minutesString);
+            }
+
+            using (var context = new ApplicationDbContext(this.ContextOptions))
+            {
+                var result = context.Tests.SingleOrDefault(t => t.Id == test.Id);
+                result.StartDate.Should().BeCloseTo(DateTime.Now, 1000);
+                result.EndDate.Should().BeCloseTo(DateTime.Now.AddMinutes(minutes), 1000);
+            }
+        }
+
+        private static AdminService PrepareSUT(ApplicationDbContext context)
+        {
+            var mapper = new MapperConfiguration(cfg => cfg.AddMaps(typeof(TestProfile))).CreateMapper();
+            var testRepository = new TestRepository(context);
+            var testAnswerRepository = new UserTestAnswersRepository(context);
+            var stringHasher = new StringHasher(new HashParameters(100, new byte[] { 1, 2 }));
+            return new AdminService(testRepository, testAnswerRepository, mapper, stringHasher);
+        }
+
+        private static Test GetTest(int number = 2) => GetTestList().First(t => t.Number == number);
+
+        private static List<Test> GetTestList() => new List<Test>()
```

Comment:
maybe ass Shared static class with these methods? I see that you are duplicating that code in other tests